tests/pgssl_test.go

```diff
@@ -0,0 +1,70 @@
+package tests
+
+import (
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+)
+
+func TestPgSSL(t *testing.T) {
+	t.Parallel()
+
+	var (
+		secretName = pointer.String("pg-certs")
+		cv         = &CoderValues{
+			Postgres: &PostgresValues{
+				Default:        &PostgresDefaultValues{Enable: pointer.Bool(false)},
+				Host:           pointer.String("1.1.1.1"),
+				Port:           pointer.String("5432"),
+				User:           pointer.String("postgres"),
+				Database:       pointer.String("postgres"),
+				PasswordSecret: pointer.String("pg-pass"),
+				SSLMode:        pointer.String("require"),
+				SSL: &PostgresSSLValues{
+					CertSecret: &CertsSecretValues{
+						Name: secretName,
+						Key:  pointer.String("cert"),
+					},
+					KeySecret: &CertsSecretValues{
+						Name: secretName,
+						Key:  pointer.String("key"),
+					},
+					RootCertSecret: &CertsSecretValues{
+						Name: secretName,
+						Key:  pointer.String("rootcert"),
+					},
+				},
+			},
+		}
+
+		objs   = LoadChart(t).MustRender(t, cv)
+		coderd = MustFindDeployment(t, objs, "coderd")
+	)
+
+	for _, vol := range []string{"pgcert", "pgkey", "pgrootcert"} {
+		AssertVolume(t, coderd.Spec.Template.Spec.Volumes, vol, func(t testing.TB, v corev1.Volume) {
+			require.NotNil(t, v.Secret)
+			assert.Equal(t, "pg-certs", v.Secret.SecretName)
```

Comment:
just curious: why assert instead of require here?

also, without a message that indicates the current `vol`, it's harder to debug this - consider using Equalf and adding a message with the vol, like: `assert.Equalf(t, "pg-certs", v.Secret.SecretName, "certs do not match for volume %s", vol)`